c7n/utils.py

```diff
@@ -488,3 +489,24 @@ def format_string_values(obj, *args, **kwargs):
         return obj.format(*args, **kwargs)
     else:
         return obj
+
+
+def init_deprecation_warnings(options):
+    def _deprecation_message(message, category, filename, lineno, file=None, line=None):
+        return '%s: %s' % (category.__name__, message)
+
+    logging.captureWarnings(True)
+    warnings.formatwarning = _deprecation_message
+    deprecation = getattr(options, 'deprecation', None)
+    action = 'error' if deprecation == 'error' else 'always'
```

Comment:
while i like the usage here, the error behavior here seems less than ideal, in that for a large collection of policies, it will feel like stop and go on correcting errors.